utils/v2tofhirr4/mappings.bal

```diff
@@ -24,40 +25,39 @@ import ballerinax/health.hl7v26;
 import ballerinax/health.hl7v27;
 import ballerinax/health.hl7v28;
 import ballerinax/health.fhir.r4 as r4;
-import ballerina/log;
 
 # Parse a string to an HL7 message.
 #
 # + msg - HL7 message as a string
 # + return - hl7v2:Message
-public function stringToHl7(string msg) returns hl7:Message|error {
-    hl7:Message hl7msg = check hl7:parse(msg);
-    return hl7msg;
-}
+public isolated function stringToHl7(string msg) returns hl7:Message|error => check hl7:parse(msg);
 
 # Transform an HL7 message to FHIR.
 #
 # + hl7 - HL7 message as a string or an hl7v2:Message  
 # + customMapper - Custom mapper implementation
 # + return - FHIR Bundle as a json
-public function v2ToFhir(string|hl7:Message hl7, SegmentToFhir? customMapper = ()) returns json|error {
+public isolated function v2ToFhir(string|hl7:Message hl7, V2SegmentToFhirMapper? customMapper = ()) returns json|error {
     hl7:Message hl7msg;
     if (hl7 is string) {
         hl7msg = check stringToHl7(hl7);
     } else {
         hl7msg = hl7;
     }
-    SegmentToFhir mapperImpl = getMapperContext().getDefaultImpl();
-    if customMapper is () {
-        return check transformToFhir(hl7msg, mapperImpl);
+    V2SegmentToFhirMapper mapper;
+    lock {
+        mapper = defaultMapper.clone();
+    }
+    if customMapper == () {
+        return transformToFhir(hl7msg, mapper.cloneReadOnly());
     }
     foreach string key in customMapper.keys() {
         if customMapper.get(key) != () {
```

Comment:
Can this be nil?

Reply:
User can input a nil for a record field, in those cases we are loading the default impl

Reply:
All the individual fields are optional fields, right? Then the value can't be nil?

`V2SegmentToFhirMapper` is open by anydata, so rest fields may be nil. Is that intentional? Can't we make it open by `function` instead?

Reply:
Yes no need of the check. fixed